packages/transaction-controller/CHANGELOG.md

```diff
@@ -7,6 +7,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [25.2.1]
+
+### Changed
+
+- Bump `TypeScript` version to `~4.9.5` ([#4084](https://github.com/MetaMask/core/pull/4084))
+
+### Fixed
+
+- Emit finished event for custodial transactions when updating status to `submitted` or `failed` ([#4092](https://github.com/MetaMask/core/pull/4092))
```

Comment:
Upon reviewing this PR, I'm noticing that the event emitted is internal and consumers won't be able to listen to it. What consumer-facing effect does emitting this event have?

Reply:
You are right it's all internal, only what is going to happen after resolving the promise is what will affect customers.
In a nutshell, this event is required to resolve the promise created by `waitForTransactionFinished`  in the `processApproval` method, and without the event, the custodial transaction won't go to `confirmed` status and the transaction will be "stuck"  in the customer perspective.

Reply:
Gotcha. I was asking because changelogs are intended for consumers first and developers second, so if we are making a reference to something that consumers can't access like an internal event, it would be better to replace that with an explanation of the functional result. Something to keep in mind for future changelog updates :)